common/gnome-shell/3.36/sass/_common.scss

```diff
@@ -2433,20 +2433,24 @@ StScrollBar {
 //
 .user-icon {
   background-size: contain;
-  color: $fg_color;
+  color: $osd_fg_color;
   border-radius: 99px;
+
+  .login-dialog-user-list-item:selected & { color: $selected_fg_color; }
+
   &:hover {
-    color: transparentize($fg_color, 0.3);
+    color: if($variant != 'lighter', lighten($osd_fg_color, 25%), darken($osd_fg_color, 12%));
   }
 
   & StIcon {
-    background-color: transparentize($osd_fg_color, 0.95);
+    background-color: transparentize($osd_fg_color, 0.75);
```

Comment:
This needs additional separate styling for `.login-dialog-user-list-item:selected .user-icon` with transparentized `$selected_fg_color`.

Reply:
Sorry, not sure I understand what you mean with this comment